File: src/models/ResearchRequest.ts
export class ResearchRequest {
    sourceType: string;
    urls: string[];
    docs: Document;
    rawText: string;

    constructor() {
        this.sourceType = '';
        this.urls = [];
        this.docs = new Document();
        this.rawText = '';
    }
}

export class  Document{
    docName: string;
    docData: string;
    docType: string;

    constructor() {
        this.docName = '';
        this.docData = '';
        this.docType = '';
    }
}

File: Backend/app/Schemas/research_request.py
import pydantic

from .document import Document
from app.Enum.content_type import ContentType


class ResearchRequest(pydantic.BaseModel):
    sourceType: ContentType
    urls: list[str] = pydantic.Field(default_factory=list)
    docs: Document | None = None
    rawText: str = ""
    context: str = ""
